group.go

```diff
@@ -134,6 +136,176 @@ func (client *Client) ListGroups() ([]Group, error) {
+				} `graphql:"descendantTeams(after: $after, first: $first)"`
+			} `graphql:"group(id: $group)"`
+		}
+	}
+	if g.Id == nil {
+		return nil, fmt.Errorf("failed to request 'DescendantTeams' for group because there is not a valid id '%s'", g.Id)
+	}
+	v := PayloadVariables{
+		"group": g.Id,
+		"first": client.pageSize,
+		"after": g.DescendantTeamsConnection.PageInfo.Start,
+	}
+	output := []Team{}
+	if err := client.Query(&q, v); err != nil {
+		return nil, err
+	}
+	output = append(output, q.Account.Group.DescendantTeams.Nodes...)
+	for q.Account.Group.DescendantTeams.PageInfo.HasNextPage {
+		v["after"] = q.Account.Group.DescendantTeams.PageInfo.End
+		if err := client.Query(&q, v); err != nil {
+			return nil, err
+		}
+		output = append(output, q.Account.Group.DescendantTeams.Nodes...)
+	}
+	return output, nil
+}
+
+func (g *Group) DescendantRepositories(client *Client) ([]Repository, error) {
+	var q struct {
+		Account struct {
+			Group struct {
+				DescendantRepositories struct {
+					Nodes    []Repository
+					PageInfo PageInfo
+				} `graphql:"descendantRepositories(after: $after, first: $first)"`
+			} `graphql:"group(id: $group)"`
+		}
+	}
+	if g.Id == nil {
+		return nil, fmt.Errorf("failed to request 'DescendantRepositories' for group because there is not a valid id '%s'", g.Id)
+	}
+	v := PayloadVariables{
+		"group": g.Id,
+		"first": client.pageSize,
+		"after": g.DescendantRepositoriesConnection.PageInfo.Start,
+	}
+	output := []Repository{}
+	if err := client.Query(&q, v); err != nil {
+		return nil, err
+	}
+	output = append(output, q.Account.Group.DescendantRepositories.Nodes...)
+	for q.Account.Group.DescendantRepositories.PageInfo.HasNextPage {
+		v["after"] = q.Account.Group.DescendantRepositories.PageInfo.End
+		if err := client.Query(&q, v); err != nil {
+			return nil, err
+		}
+		output = append(output, q.Account.Group.DescendantRepositories.Nodes...)
+	}
+	return output, nil
+}
+
+func (g *Group) DescendantServices(client *Client) ([]Service, error) {
+	var q struct {
+		Account struct {
+			Group struct {
+				DescendantServices struct {
+					Nodes    []Service
+					PageInfo PageInfo
+				} `graphql:"descendantServices(after: $after, first: $first)"`
+			} `graphql:"group(id: $group)"`
+		}
+	}
+	if g.Id == nil {
+		return nil, fmt.Errorf("failed to request 'DescendantServices' for group because there is not a valid id '%s'", g.Id)
+	}
+	v := PayloadVariables{
+		"group": g.Id,
+		"first": client.pageSize,
+		"after": g.DescendantServicesConnection.PageInfo.Start,
+	}
+	output := []Service{}
+	if err := client.Query(&q, v); err != nil {
+		return nil, err
+	}
+	output = append(output, q.Account.Group.DescendantServices.Nodes...)
+	for q.Account.Group.DescendantServices.PageInfo.HasNextPage {
+		v["after"] = q.Account.Group.DescendantServices.PageInfo.End
+		if err := client.Query(&q, v); err != nil {
+			return nil, err
+		}
+		output = append(output, q.Account.Group.DescendantServices.Nodes...)
```

Comment:
We may need to call Hydrate on every service object because it has sub resources with pagination.

We may want to consider changing this to return `ServiceId` rather then a `Service` object which then forces the user to make additional API calls to turn a ServiceId into a full Service object with all the pagination

Reply:
Do you mean as a user, they would run the subcommand and get something similar to a list, e.g. Name, Alias, ID?  Then they would run `get service <ID>` after to get more info on whichever service they're interested in?

Reply:
ya exactly - that way we avoid infinite recursion